impeller/renderer/backend/vulkan/blit_command_vk.cc

```diff
@@ -0,0 +1,178 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/blit_command_vk.h"
+
+#include "impeller/renderer/backend/vulkan/commands_vk.h"
+#include "impeller/renderer/backend/vulkan/texture_vk.h"
+
+namespace impeller {
+
+BlitEncodeVK::~BlitEncodeVK() = default;
+
+// BEGIN: BlitCopyTextureToTextureCommandVK
+
+BlitCopyTextureToTextureCommandVK::~BlitCopyTextureToTextureCommandVK() =
+    default;
+
+std::string BlitCopyTextureToTextureCommandVK::GetLabel() const {
+  return label;
+}
+
+[[nodiscard]] bool BlitCopyTextureToTextureCommandVK::Encode(
+    FencedCommandBufferVK* fenced_command_buffer) const {
+  // cast source and destination to TextureVK
+  auto& source_tex_vk = TextureVK::Cast(*source);
+  auto& dest_tex_vk = TextureVK::Cast(*destination);
+
+  // get the vulkan image and image view
+  auto source_image = source_tex_vk.GetImage();
+  auto dest_image = dest_tex_vk.GetImage();
+
+  // copy the source image to the destination image, from source_region to
+  // destination_origin.
+  vk::ImageCopy image_copy{};
```

Comment:
IIRC, all vulkan.hpp objects have a default constructor. So you could skip the {}. Or rather `auto imageCopy = vk::ImageCopy{};` instead.